examples/step-40/step-40.cc

```diff
@@ -44,7 +44,7 @@
 // @endcode
 //
 // Using this logic, the following lines will then import either the
-// PETSc or Trilinos wrappers into the namespace `LA` (for "linear
+// PETSc or Trilinos wrappers into the namespace `LA` (for linear
```

Comment:
It turns out the situation is less sever than I thought it would be. Here, the quote character is unmatched in the entire document and this trips up doxygen. Which, admittedly, isn't nice. doxygen could fail gracefully. But this is also syntactically wrong - so let's fix the single quote issue to make doxygen happy.

Note, all other quote-matching issues we encountered were for keywords, for example
```
  // @ref GlossEntry "description that spans more than
  // one line"
```
These quotes have to be on a single line. But this is something easy to `grep` for.